scripts/local_cloudbuild_test.py

```diff
@@ -76,17 +76,89 @@ def test_validate_arg_regex(self):
             local_cloudbuild.validate_arg_regex('abc', re.compile('a[d]c'))
 
 
+    def test_validate_arg_dict(self):
+        valid_cases = (
+            # Normal case, field present and correct type
+            ('', {}),
+            ('_A=1', {'_A':'1'}),
+            ('_A=1,_B=2', {'_A':'1', '_B':'2'}),
+            # Repeated key is ok
+            ('_A=1,_A=2', {'_A':'2'}),
+            # Extra = is ok
+            ('_A=x=y=z,_B=2', {'_A':'x=y=z', '_B':'2'}),
+            # No value is ok
+            ('_A=', {'_A':''}),
+        )
+        for valid_case in valid_cases:
+            with self.subTest(valid_case=valid_case):
+                s, expected = valid_case
+                self.assertEqual(
+                    local_cloudbuild.validate_arg_dict(s),
+                    expected)
+
+        invalid_cases = (
+            # No key
+            ',_A',
+            '_A,',
+            # Invalid variable name
+            '_Aa=1',
+            '_aA=1',
+        )
+        for invalid_case in invalid_cases:
+            with self.subTest(invalid_case=invalid_case):
+                with self.assertRaises(argparse.ArgumentTypeError):
+                    local_cloudbuild.validate_arg_dict(invalid_case)
+
+
 class LocalCloudbuildTest(unittest.TestCase):
 
     def setUp(self):
         self.testdata_dir = 'testdata'
         assert os.path.isdir(self.testdata_dir), 'Could not run test: testdata directory not found'
 
+    def test_sub_and_quote(self):
+        valid_cases = (
+            # Empty string
+            ('', {}, "''"),
+            # No substitutions
+            ('a', {}, 'a'),
+            # Unused substitutions
+            ('a', {'FOO':'foo'}, 'a'),
+            # Defined builtin substitution
+            ('a$FOOb', {'FOO':'foo'}, 'afoob'),
+            ('a${FOO}b', {'FOO':'foo'}, 'afoob'),
+            # Undefined builtin substitution
+            ('a$FOOb', {}, 'ab'),
+            ('a${FOO}b', {}, 'ab'),
+            # Defined user substitution
+            ('a$_FOOb', {'_FOO':'_foo'}, 'a_foob'),
+            ('a${_FOO}b', {'_FOO':'_foo'}, 'a_foob'),
+            # Multiple substitutions
+            ('$FOO${FOO}${BAR}$FOO', {'FOO':'foo', 'BAR':'bar'}, 'foofoobarfoo'),
+        )
+        for valid_case in valid_cases:
+            with self.subTest(valid_case=valid_case):
+                s, subs, expected = valid_case
+                actual = local_cloudbuild.sub_and_quote(s, subs)
+                self.assertEqual(actual, expected)
+
+        invalid_cases = (
+            # Undefined user substitution
+            ('a$_FOOb', {}),
+            ('a${_FOO}b', {}),
```

Comment:
I was going to be lazy here, but it's nice to have and should be a nice way to catch typos in cloudbuild files.  So I added checking code and tests.  The implementation is very slightly ugly since I have an in-out parameter that I pass around and modify.